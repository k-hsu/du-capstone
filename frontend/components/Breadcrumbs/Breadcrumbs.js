import React from "react";
import Link from "../Link/Link";
import Flex from "../Flex/Flex";
import Text from "../Text/Text";
import { color, fontWeight, spacing } from "../../theme";

const Breadcrumbs = ({ path = [], currentPage }) => {
  return (
    <Flex gap={spacing["0.5"]}>
      {path.map((pathProps, index) =>
        [
          <Text key={pathProps.id}>
            <Link
              href={pathProps.href}
              color={color.black}
              textDecoration="none"
            >
              {pathProps.page}
            </Link>
          </Text>,
          <Text key={`div-${index}`}>/</Text>,
        ].flatMap((element) => element)
      )}
      <Text fontWeight={fontWeight.bold}>{currentPage}</Text>
    </Flex>
  );
};

export default Breadcrumbs;
