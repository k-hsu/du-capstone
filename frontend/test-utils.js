import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderHook } from "@testing-library/react-hooks";

function setup(jsx) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}

export * from "@testing-library/react";
export { renderHook, setup };
