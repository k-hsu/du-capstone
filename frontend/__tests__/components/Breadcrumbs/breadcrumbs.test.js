import React from "react";
import { render, screen, renderer } from "../../../test-utils";
import Breadcrumbs from "../../../components/Breadcrumbs/Breadcrumbs";

describe("Breadcrumbs", () => {
  it("should render breadcrumbs component", () => {
    render(
      <Breadcrumbs
        path={[{ id: "crumb-path", page: "crumb", href: "/to-the-crumb" }]}
        currentPage="bread"
      />
    );
    expect(screen.getByText("bread")).toBeInTheDocument();
    expect(screen.getByText("crumb")).toBeInTheDocument();
    expect(screen.getByText("crumb")).toHaveAttribute("href", "/to-the-crumb");
  });
  it("should render breadcrumbs component styles", () => {
    const component = renderer.create(
      <Breadcrumbs
        path={[{ id: "crumb-path", page: "crumb", href: "/to-the-crumb" }]}
        currentPage="bread"
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
