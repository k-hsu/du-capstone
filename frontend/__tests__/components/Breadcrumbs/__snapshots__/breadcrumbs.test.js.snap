// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should render breadcrumbs component styles 1`] = `
<div
  className="css-12u5n7l"
  direction="row"
>
  <p
    className="css-ms6223"
    fontFamily="IBM Plex Sans"
  >
    <a
      className="css-1486csp"
      color="#000"
      fontFamily="IBM Plex Sans"
      href="/to-the-crumb"
      textDecoration="none"
    >
      crumb
    </a>
  </p>
  <p
    className="css-ms6223"
    fontFamily="IBM Plex Sans"
  >
    /
  </p>
  <p
    className="css-1v2p52x"
    fontFamily="IBM Plex Sans"
    fontWeight="700"
  >
    bread
  </p>
</div>
`;
